<robot name="shadow_arm">

  <link name="shadowarm_base">
    <inertial>
      <origin xyz="0 0 0.005" rpy="0 0 0"/> 
      <mass value="0.5" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0 0 0.005" rpy="1.57079632679 0 0"/> 
      <geometry name="shadowarm_base_visual">
        <mesh filename="package://sr_hand/model/meshes/arm/arm_base.mesh" scale="0.001 0.001 0.001" />
      </geometry>
      <material name="shadowarm_base_material">
	<color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.005" rpy="0 0 0" /> 
      <geometry name="shadowarm_base_collision">
        <box size="0.03 0.03 0.01"/>
      </geometry>
    </collision>
  </link>
  
  <link name="shadowarm_trunk">
    <inertial>
      <origin xyz="0 0 0.27025" rpy="0 0 0" />
      <mass value="5" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0 0 0.0165" rpy="1.57079632679 3.14159265359 1.57079632679" />
      <geometry name="shadowarm_trunk_visual">
	<mesh filename="package://sr_hand/model/meshes/arm/arm_trunk.mesh" scale="0.001 0.001 0.001" />
      </geometry>
      <material name="shadowarm_trunk_material">
	<color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0.27025" rpy="0 0 0"/>
      <geometry name="shadowarm_trunk_collision">
	<box size="0.03 0.03 0.5405"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="trunk_rotation" type="revolute">
    <parent link="shadowarm_base"/>
    <child link="shadowarm_trunk"/>   
    <origin xyz="0 0 0.01" rpy="0 0 0" />
    <axis xyz="0 0 1" />
    <limit lower="-1.57079632679" upper="0.785398163397"
    effort="10" velocity="1.0"/>
  </joint>
  
  <link name="shadowarm_upperarm">
    <inertial>
      <origin xyz="0.1845 -0.067 0" rpy="0 1.571 0" />
      <mass value="5" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0 0.067 0" rpy="-1.571 0 -1.571" />
      <geometry name="shadowarm_upperarm_visual">
	<mesh filename="package://sr_hand/model/meshes/arm/upper_arm.mesh" scale="0.001 0.001 0.001" />
      </geometry>
      <material name="shadowarm_upperarm_material">
	<color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.1845 0.067 0" rpy="1.571 1.571 0"/>
      <geometry name="shadowarm_upperarm_collision">
	<box size="0.03 0.03 0.369"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="shoulder_rotation" type="revolute">
    <parent link="shadowarm_trunk"/>
    <child link="shadowarm_upperarm"/>   
    <origin xyz="0 -0.134 0.5405" rpy="0 0 0" />
    <axis xyz="0 1 0" />
    <limit lower="0" upper="1.57079632679"
    effort="10" velocity="1.0"/>
  </joint>
  
  <link name="shadowarm_lowerarm">
    <inertial>
      <origin xyz="0.0655 0 -0.08" rpy="0 1.571 0" />
      <mass value="5" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0.036 0 -0.03" rpy="3.142 0.5 0"/>
      <geometry name="shadowarm_lowerarm_visual">
	<mesh filename="package://sr_hand/model/meshes/arm/elbow_flex.mesh" scale="0.001 0.001 0.001" />
      </geometry>
      <material name="shadowarm_lowerarm_material">
	<color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0655 0 -0.08" rpy="0 1.571 0"/>
      <geometry name="shadowarm_lowerarm_collision">
	<box size="0.03 0.03 0.131"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="elbow_abduction" type="revolute">
    <parent link="shadowarm_upperarm"/>
    <child link="shadowarm_lowerarm"/>   
    <origin xyz="0.369 0 0" rpy="0 0 0" />
    <axis xyz="0 -1 0" />
    <limit lower="0" upper="2.09439510239"
    effort="10" velocity="1.0"/>
  </joint>
  
  <link name="shadowarm_handsupport">
    <inertial>
      <origin xyz="0 0 0" rpy="0 1.571 0" />
      <mass value="1" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="-0.04 0 0" rpy="0 0 -1.571"/>
      <geometry name="shadowarm_handsupport_visual">
	<mesh filename="package://sr_hand/model/meshes/arm/elbow_rotate_motor.mesh" scale="0.001 0.001 0.001" />
      </geometry>
      <material name="shadowarm_handsupport_material">
	<color rgba="0.6 0.6 0.6 1"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 1.571 0"/>
      <geometry name="shadowarm_handsupport_collision">
	<cylinder radius="0.06" length="0.001"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="forearm_rotation" type="revolute">
    <parent link="shadowarm_lowerarm"/>
    <child link="shadowarm_handsupport"/>   
    <origin xyz="0.131 0 -0.08" rpy="0 0 0" />
    <axis xyz="1 0 0" />
    <limit lower="-1.57079632679" upper="1.57079632679"
    effort="10" velocity="1.0"/>
  </joint>

</robot>